Documentation/Multiple Selections.md

```diff
@@ -0,0 +1,96 @@
+### Multiple Carets / Multiple Cursors / Multiple Selections
+
+If the host supports multiple selections, VsVim will also support them.
+
+#### Architecture
+
+Most parts of VsVim are unware of multiple selections. In order to keep things
+simple and preserve the single selection model for most of VsVim, there are two
+major components that handle multiple selections:
+
+- Running mode commands for all selections (`ICommonOperations.RunForAllSelections`)
+- The multi-selection tracker (`MultiSelectionTracker`)
+
+For commands where it makes sense to run them for each caret or selection, the
+corresponding command runner will wrap the running of a single command into a
+batch execution for each selection using `RunForAllSelections`. Then, while
+running an individual command, the multi-selection infrastructure will:
+
+- Set a single temporary selection, including a caret position
+- Run the command as if there were a single selection
+- Determine the desired resulting selection from the command result
+
+When all the individual commands have run, the accumulated selections will be
+set all at once. This causes the non-multi-selection aware code to still see
+the primary selection as the real and only selection, while maintaining the
+secondary selections behind the scenes.
+
+#### The Multi-Selection API
+
+Unfortunately, not all versions of the editor API support multiple selections.
+In order to handle this, there are two API entry points in the Vim host:
+
+- `IVimHost.GetSelectedSpans` - member
+- `IVimHost.SetSelectedSpans` - member
+
+A selected span is a triple of caret point, anchor point, and active point.
+All three values are virtual snapshot points.  For multiple caret support,
+a caret is just a zero-width selection. Selected spans are associated with
+a specific text view, just like the normal caret and normal selection.
+
+If there is no multi-selection support, getting the selected spans always
+returns a single (possibly empty) span and setting multiple selected spans
+ignores all of the secondary selections.
+
+If there is multi-selection support, then the first selected span is the
+primary selection and any subsequent selected spans are secondary selections,
+sorted in ascending order by position within the text buffer.
+
+In order to insulate the code base from the host, all consumers who use the
+multi-selection API access the selections using related APIs in the common
+operations module:
+
+- `ICommonOperations.SelectedSpans` - property
+- `ICommonOperations.SetSelectedSpans` - member
+- `ICommonOperations.SelectedSpansSet` - event
+
+#### VsVim Key Bindings
+
+- `<C-A-LeftMouse>` - add or remove caret (normal, visual, select, insert)

```

Comment:
Are the modes in the parens listing where the key bindings are valid?

Reply:
Yes, I should clarify that.